tests/firedrake/multigrid/test_p_multigrid.py

```diff
@@ -73,6 +73,17 @@ def test_reconstruct_degree(tp_mesh, mixed_family):
         assert e == PMGPC.reconstruct_degree(elist[0], degree)
 
 
+def test_work_function_cache(tp_mesh):
+    from firedrake.preconditioners.pmg import StandaloneInterpolationMatrix
+
+    V1 = FunctionSpace(tp_mesh, "Lagrange", 1)
+    V2 = FunctionSpace(tp_mesh, "Lagrange", 1)
+    assert V1 is not V2
+    w1 = StandaloneInterpolationMatrix.work_function(V1)
+    w2 = StandaloneInterpolationMatrix.work_function(V2)
+    assert w1 is w2
```

Comment:
This make me wonder if we should implement `restore()` method to keep track of if the `work_function` is `in_use` or not, but it's not the scope of this PR.

Reply:
Should we also look at `FunctionSpace.get_work_function()`?

Reply:
That's probably better as they already have `restore` method.

Reply:
This will soon become irrelevant. I plan to eliminate this class in favour of `Interpolator`